drivers/gles2/shader_compiler_gles2.cpp

```diff
@@ -754,12 +754,9 @@ String ShaderCompilerGLES2::_dump_node_code(SL::Node *p_node, int p_level, Gener
 				} break;
 
 				case SL::OP_MOD: {
-
-					code += "mod(float(";
-					code += _dump_node_code(op_node->arguments[0], p_level, r_gen_code, p_actions, p_default_actions, p_assigning);
-					code += "), float(";
-					code += _dump_node_code(op_node->arguments[1], p_level, r_gen_code, p_actions, p_default_actions, p_assigning);
-					code += "))";
+					String a = _dump_node_code(op_node->arguments[0], p_level, r_gen_code, p_actions, p_default_actions, p_assigning);
+					String n = _dump_node_code(op_node->arguments[1], p_level, r_gen_code, p_actions, p_default_actions, p_assigning);
+					code += "(" + a + " - " + n + " * (" + a + " / " + n + "))";
```

Comment:
Should probably add a `floor()` in there to match the official spec: https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/mod.xhtml

Reply:
Well `floor(int)` won't work, but here the mod implementation will be wrong for float values.

Reply:
There'd be no point, right? As far as I understand this operation already require both values to be int.

Reply:
Correct, my bad.

BTW, the same changes are needed for `OP_ASSIGN_MOD` (`i %= n`).